### ISSUE TITLE
Customize the appearance (theming)

### ISSUE THREAD

**Issue by mittalyashu** [netlify/netlify-cms#1727]
**Is your feature request related to a problem? Please describe.**

No. It is a feature request.

**Describe the solution you'd like**

![2018-09-07_13-55-02](https://user-images.githubusercontent.com/29014463/45207537-d7858580-b2a5-11e8-92b9-a10fa254d553.gif)

**Additional context**

I have also discussed about this in the NetlifyCMS community talk on 29-08-2018.

**Comment by rchrdnsh** [comment]
I would love to be able to do this so that I can give clients a custom themed CMS to edit their content, which would look very professional and provide a premium experience.

**Comment by rikurb8** [comment]
Hey guys,

any news on this / is there discussion happening about this somewhere else?

**Comment by tomrutgers** [comment]
There hasn't been much chatter about this topic lately @rikurb8 
Although the maintainers would love to see a feature like this implemented, there are still more pressing matters to work on. We're happy to support any effort to help this thing to the finish line though!

**Comment by erquhart** [comment]
I've long thought we needed to split the UI components out so they could be replaced with custom ones, but I bet what people really want is just to change colors and maybe fonts? Most of this is already controlled by a set of variables, so in the future it could be feasible to theme through the CMS config: https://github.com/netlify/netlify-cms/blob/ccef446d72f96fa8d8db9bf2d0be5633eff79979/packages/netlify-cms-ui-default/src/styles.js#L4-L82

**Comment by rchrdnsh** [comment]
colors, fonts, logos, light and dark mode, that kind of stuff, yes. Basically so I can give the client a 'custom' CMS for thier custom site/app/thing that has their logo, colors, fonts, etc...

**Comment by erquhart** [comment]
The only complicated part is doing this in a way that works for extensions, namely widgets.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rchrdnsh** [comment]
yeah, looks pretty str84ward :-)

**Comment by numoonchld** [comment]
> I've long thought we needed to split the UI components out so they could be replaced with custom ones, but I bet what people really want is just to change colors and maybe fonts? Most of this is already controlled by a set of variables, so in the future it could be feasible to theme through the CMS config:
> 
> https://github.com/netlify/netlify-cms/blob/ccef446d72f96fa8d8db9bf2d0be5633eff79979/packages/netlify-cms-ui-default/src/styles.js#L4-L82

How do I go about accessing these variables in the index.html of the CMS UI? I tried adding a script HTML element in the head and changing the colors assigned but nothing happened.

https://github.com/numoonchld/nmc-ossu-bi-ncms/blob/master/admin/index.html

**Comment by erquhart** [comment]
They're internal, no outside access at the moment.

**Comment by numoonchld** [comment]
Alright. Will watch for updates.

**Comment by asaldivar** [comment]
👍 for this feature

**Comment by omfgnuts** [comment]
Any progress on this issue?

**Comment by flacito** [comment]
Consider the way tools like VuePress do themes:  https://vuepress.vuejs.org/theme/using-a-theme.html#using-a-theme-from-a-dependency.

A very small simple set of files and directory structure leads to great ways to totally customize the layout and look of the site.

**Comment by kenkoy** [comment]
Any update/progress?

**Comment by erezrokah** [comment]
Hi @kenkoy, the CMS is a community based project. I you'd like to contribute to this issue, it might be better to follow https://github.com/netlify/netlify-cms/issues/2557

**Comment by leo60228** [comment]
As a workaround, you can add this to the end of `<head>`:
```html
<script src="https://cdn.jsdelivr.net/npm/darkreader@4.9.27/darkreader.min.js"></script>
<script>
  DarkReader.enable();
</script>
<style>
html, body {
  width: 100%;
  height: 100%;
}
</style>
```

**Comment by ecupaio** [comment]
The color contrasts are all inaccessible. I have perfect vision and the gray on gray is hard to read for me.

**Comment by corbs** [comment]
This would be such a fantastic feature to have!

**Comment by patoperpetua** [comment]
## Proposal: incremental design tokens / UI theming on `main`

Hi maintainers 👋 — we'd like to contribute theming support and wanted alignment before opening a PR.

### Motivation
Many projects want to brand the CMS UI (primary color, surfaces, typography) to match their design system. Today colors/fonts live as JS constants in `packages/decap-cms-ui-default/src/styles.js` and are baked into Emotion styles, so they aren't overridable via `config.yml` or CSS.

### Prior art (as we understand it)
- This issue (#1727) — long-standing theming request
- #3559 — detailed palette / `config.yml` theming proposal
- #3266 / #3267 — internal theme + dark mode
- #7101 — new UI draft with `branding.theme` + `CMS.registerTheme()` (closed, not merged)
- `next-ui` + #7138 — Emotion light/dark themes and shade scales (branch appears stalled / behind `main`)

### Proposed approach (MVP on current UI)
Rather than waiting on a full `next-ui` port, start with a **small, backward-compatible token layer** on the existing UI:

1. **Define a public semantic token set** sourced from the existing `styles.js` values, e.g.:
   - `color.primary` / `color.active`
   - `color.background` / `color.foreground` / `color.text` / `color.textLead`
   - status colors (draft / review / ready)
   - `font.family` (optional in MVP)
   - `radius` / border radius (optional in MVP)
2. **Expose tokens as CSS custom properties** (and/or Emotion theme) so components read tokens instead of hard-coded hex where practical.
3. **Allow overrides** via:
   - `config.yml` under something like `branding.theme` (aligned with ideas from #7101 / #3559)
   - optionally `CMS.registerTheme(name, theme)` for programmatic use
4. **Defaults unchanged** — no visual change unless overrides are provided.
5. **Docs + minimal Storybook/demo** showing 2–3 override examples.

#### Example config sketch
```yml
branding:
  # existing logo support can stay as-is
  theme:
    primary: '#112f4e'
    background: '#f5f6f8'
    text: '#313d3e'
    status:
      draft: '#70399f'
      in_review: '#754e00'
      ready: '#005614'
```

#### Non-goals for MVP
- Full redesign / dark mode (can follow once tokens exist; related: #3267)
- Replacing every hard-coded color in all widgets in the first PR
- A full Style Dictionary / DTCG pipeline
- Landing the entire `next-ui` branch

### Questions for maintainers
1. Is **targeting `main` / current UI** acceptable for an MVP, or should theming wait for / land on `next-ui`?
2. Prefer **CSS variables**, **Emotion `ThemeProvider`**, or both?
3. Is `branding.theme` the right config namespace, or do you prefer `theme:` at the root / something else?
4. Would you like this scoped as a series of small PRs (token plumbing → config wiring → docs), or one medium PR?

Happy to open a draft PR once we have a green light on direction. Thanks!

**Comment by patoperpetua** [comment]
Follow-up to the proposal above: opened a **draft spike PR** so there’s something concrete to react to:

https://github.com/decaporg/decap-cms/pull/7985

It wires semantic colors to CSS custom properties and accepts optional `branding.theme` overrides, with defaults unchanged when omitted. Happy to reshape the token list / config shape based on maintainer feedback before expanding coverage.
